src/main/java/org/testng/internal/ITestClassConfigInfo.java

```diff
@@ -0,0 +1,33 @@
+package org.testng.internal;
+
+import org.testng.ITestNGMethod;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public interface ITestClassConfigInfo {
+    /**
+     * get the relationship between instance and class level config methods
+     * @return relationship between instance and class level config methods
+     */
+    default Map<String, ITestNGMethod[]> getBeforeClassConfig() {
+        return new HashMap<>();
+    }
+
+    /**
+     * get all before class config methods
+     * @return all before class config methods
+     */
+    default ITestNGMethod[] getAllBeforeClassMethods() {
```

Comment:
For historical reasons, TestNG uses a lot of arrays but collections are more practice. Could you use `List`s instead of arrays?

Reply:
OK, all methods have change to List.